modules/caddyhttp/reverseproxy/reverseproxy.go

```diff
@@ -273,15 +274,26 @@ func (h *Handler) Provision(ctx caddy.Context) error {
 		}
 
 		// if active health checks are enabled, configure them and start a worker
-		if h.HealthChecks.Active != nil &&
-			(h.HealthChecks.Active.Path != "" || h.HealthChecks.Active.Port != 0) {
+		if h.HealthChecks.Active != nil && (h.HealthChecks.Active.Path != "" ||
+			h.HealthChecks.Active.Uri != "" ||
+			h.HealthChecks.Active.Port != 0) {
+
 			h.HealthChecks.Active.logger = h.logger.Named("health_checker.active")
 
 			timeout := time.Duration(h.HealthChecks.Active.Timeout)
 			if timeout == 0 {
 				timeout = 5 * time.Second
 			}
 
+			// parse the Uri string (supports path and query)
+			if h.HealthChecks.Active.Uri != "" {
+				url, err := url.Parse(h.HealthChecks.Active.Uri)
```

Comment:
```suggestion
				parsedURI, err := url.Parse(h.HealthChecks.Active.Uri)
```

Just to avoid any confusion with the `url` package (even though scope is small).